lib/cell.rb

```diff
@@ -4,22 +4,43 @@ class Cell
   def initialize(coordinate)
     @coordinate = coordinate
     @ship = nil
+    @fired_upon = false
   end
 
   def empty?
-    @ship.nil?
+    if @ship == nil
+      true
+    else
+      false
+    end
   end
 
   def place_ship(ship)
     @ship = ship
   end
 
   def fired_upon?
-    @ship.health < @ship.length
+    @fired_upon
   end
 
   def fire_upon
-    @ship.hit
+    @fired_upon = true
+    if empty? == false
+      @ship.hit
+    end
   end
 
+  def render(show_ship = false)
+    if fired_upon? == true && empty? == true
+      "M"
+    elsif fired_upon? == true && @ship.sunk? == true && empty? == false
+      "X"
+    elsif fired_upon? == true && empty? == false
+      "H"
+    elsif empty? == false && show_ship == true
+      "S"
+    else
+      "."
+    end
+  end
```

Comment:
you can remove all of the "== true" and "== false" to clean up the syntax. Here is how:
```
if fired_upon? && empty?
  "M"
elsif fired_upon? && @ship.sunk? && !empty?
  "X"
elsif fired_upon? && !empty?
  "H"
elsif !empty? && show_ship
  "S"
else
  "."
end
```